### ISSUE TITLE
A dot is valid in a parameter name for WITH in CREATE MATERIALIZED VIEW

### ISSUE THREAD

**Issue by davetapley** [sqlfluff/sqlfluff#4072]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Follow [this tutorial](https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/about-continuous-aggregates/) with:

```sql

CREATE MATERIALIZED VIEW daily_average WITH (timescaledb.continuous)
    AS SELECT time_bucket('1 day', time) as Day,
              location,
              avg(temperature)
       FROM temperatures
       GROUP BY day, location;
```

Then
```
sqlfluff  parse --dialect postgres timescale.sql 
echo $?
```

### Expected Behaviour

`0`

### Observed Behaviour

```
==== parsing violations ====
L:  98 | P:   1 |  PRS | Line 98, Position 1: Found unparsable section: 'CREATE
                       | MATERIALIZED VIEW daily_average\nWITH ...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly
```

### How to reproduce

See above.

### Dialect

postgres

### Version

`1.4.2`

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by davetapley** [comment]
Because https://github.com/sqlfluff/sqlfluff/blob/a006378af8b670f9235653694dbcddd4c62d1ab9/src/sqlfluff/dialects/dialect_postgres.py#L2041-L2045

I tried adding `\.` before `$` here:
https://github.com/sqlfluff/sqlfluff/blob/a006378af8b670f9235653694dbcddd4c62d1ab9/src/sqlfluff/dialects/dialect_postgres.py#L270-L271

And adding to `test/fixtures/dialects/postgres/postgres_create_materialized_view.yml`:
```sql
CREATE MATERIALIZED VIEW my_mat_view
WITH (left.right)
AS
SELECT a
FROM my_table;
```

But `tox -e generate-fixture-yml -- --dialect postgres` fails with:

```
sqlfluff.core.errors.SQLParseError: Could not parse: test/fixtures/dialects/postgres/postgres_create_materialized_view.yml
ERROR: InvocationError for command /workspaces/sqlfluff/.tox/generate-fixture-yml/bin/python test/generate_parse_fixture_yml.py --dialect postgres (exited with code 1)
```
